slash-commands/utility/clearinfractions.js

```diff
@@ -0,0 +1,64 @@
+const {SlashCommandBuilder} = require('@discordjs/builders');
+const {sendToAuditLogsChannel} = require('../../helpers/sendToAuditLogs');
+const {promisePool} = require('../../config/db');
+
+module.exports = {
+  data: new SlashCommandBuilder()
+    .setName('clearinfractions')
+    .setDescription('Removes all infractions from a user')
+    .addUserOption((option) =>
+      option.setName('user').setDescription('The user').setRequired(true)
+    ),
+  async execute(interaction) {
+    const targetUser = await interaction.options.getUser('user');
+    try {
+      const hasInfractions = await checkForInfractions(targetUser.id);
+      if (!hasInfractions) {
+        return interaction.reply('This user does not have any infractions.');
+      }
+      await deactivateInfractionsDB(interaction, targetUser);
+      await sendToAuditLogsChannel(interaction, {
+        color: '#0099ff',
+        titleMsg: `${interaction.user.tag} cleared all infractions for ${targetUser.username}#${targetUser.discriminator}`,
+        targetUser,
+      });
+      await clearInfractionsResponse(interaction, targetUser);
+    } catch (err) {
+      console.error(err);
+      interaction.reply(`There was a problem clearing the infractions.`);
+    }
+  },
+};
+
+async function checkForInfractions(userId) {
+  const sql = `SELECT * FROM infractions WHERE user = ? AND valid = true`;
+  const values = [userId];
+  try {
+    const [infractions] = await promisePool.execute(sql, values);
+    return infractions.length ? true : false;
```

Comment:
I wonder if it would be better to select the COUNT of infractions and return that. Since we'll never have a negative value, we can use the truthiness/falsiness of the count instead of querying all the columns and checking length. If we really want to ensure a boolean we could also use `!!` to coerce it. Thoughts?

Reply:
I think using COUNT is a good idea. But I ended up returning nrOfinfractions > 0 which may be easier to understand vs !!nrOfInfractions while still avoiding type coercion.